File: problem-solving/Easy/6_staircase/javascript/main.js
// URL : https://www.hackerrank.com/challenges/staircase/problem

function staircase(n) {
  if (0 < n && n <= 100) {
    for (let i = n; i > 0; i--) {
      let string = "";
      for (let j = 1; j < n + 1; j++) {
        if (j < i) string += " ";
        else string += "#";
      }
      console.log(string);
    }
  }
}

// test
staircase(6);

File: problem-solving/Easy/6_staircase/python/main.py
# URL : https://www.hackerrank.com/challenges/staircase/problem

def staircase(n):
    if 0 < n <= 100:
        for i in range(n, 0, -1):
            string = ""
            for j in range(1, n+1):
                if j < i:
                    string += " "
                else:
                    string += "#"
            print(string)


# test
staircase(6)
